### PR TITLE
add prometheus_metrics support

### PR DESCRIPTION
**Pull request by ebattat**
Adding prometheus metrics support:
1. Run directly the query and get prometheus metrics
2. Run query yaml file and get prometheus metrics
3. Add prometheus_metrics decorator ability:

For using custom metrics file (optional), pass it as argument to decorator:
[This will collect custom metrics for run() method]

```
current_dir = os.path.dirname(os.path.abspath(__file__))
yaml_full_path = os.path.join(current_dir, 'metrics.yaml')


@prometheus_metrics(yaml_full_path=yaml_full_path)
def run():
    print('sleep 30 sec')
    time.sleep(30)
run()
```

### CODE REVIEW

File: docs/source/prometheus.md [hunk 1]
```diff
@@ -1,6 +1,36 @@
 
 ## Inspect Prometheus Metrics
 
+For Inspect Prometheus Metrics use @prometheus_metrics decorator.
+
+For using default metrics benchmark_runner.common.prometheus.metrics-default.yaml:
+[This will collect the metric for run() method]
+```
+from benchmark_runner.common.prometheus.prometeus_metircs import prometheus_metrics
```
Comment (RobertKrawitz) [inline docs/source/prometheus.md:9]: Fix when you fix the file name.

File: benchmark_runner/common/prometheus/prometeus_metircs.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+
+from functools import wraps
+
+from benchmark_runner.common.logger.logger_time_stamp import logger_time_stamp, logger
+from benchmark_runner.common.prometheus.prometheus_metrics_operations import PrometheusMetricsOperation
+
+
+def prometheus_metrics(yaml_full_path: str = None):
```
Comment (RobertKrawitz) [inline benchmark_runner/common/prometheus/prometeus_metircs.py:8]: This looks to be a very complex way of doing it, although I do like that doing this as a decorator makes use very simple.

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]
Just one minor fix to a filename.

File: benchmark_runner/common/prometheus/prometeus_metircs.py (hunk 2 above)
Comment (ebattat) [reply benchmark_runner/common/prometheus/prometeus_metircs.py:8]: Once it is in decorator, you just need to add the stamp "@prometheus_metrics" before a method.
There is option also to run it step by step:

```
    prometheus_metrics_operation = PrometheusMetricsOperation()
    prometheus_metrics_operation.init_prometheus()
    time.sleep(70)
    prometheus_metrics_operation.finalize_prometheus()
    prometheus_metrics_operation.run_prometheus_queries(query_yaml_file=yaml_full_path)
```

File: benchmark_runner/common/prometheus/prometeus_metircs.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+
```
[the remark this answers is not in the document]
Comment (ebattat) [reply benchmark_runner/common/prometheus/prometeus_metircs.py:1]: Fixed

**Review by RobertKrawitz — APPROVED** [review]
lgtm

**Pull request merged**